.github/workflows/generate-sbom.yaml

```
@@ -21,45 +47,68 @@ jobs:
       - name: Create directories
         shell: bash
         run: |
-          mkdir -p metalk8s_sbom/repo
-          mkdir -p metalk8s_sbom/sbom
+          mkdir -p ${{ env.BASE_PATH }}/repo
+          mkdir -p ${{ env.BASE_PATH }}/iso
+          mkdir -p ${{ env.SBOM_PATH }}
 
-      - name: Checkout
-        uses: actions/checkout@v4
-        with:
+      - name: Checkout repo for scanning
+        uses: actions/checkout@v4  
+        with:  
+          fetch-depth: 0  
+          fetch-tags: true
           ref: ${{ inputs.ref }}
-          path: metalk8s_sbom/repo/metalk8s
+          path: ${{ env.BASE_PATH }}/repo/metalk8s
+
+      - name: Generate sbom for repository
+        uses: scality/sbom@v1.2.0
+        with:
+          target: ${{ env.BASE_PATH }}/repo/metalk8s
+          output-dir: ${{ env.SBOM_PATH }}
 
-      - name: Scan metalk8s repository
-        uses: scality/sbom@v1.1.0
+      - name: Get artifacts URL
+        if: ${{ ! inputs.artifacts-url }}
+        uses: scality/action-artifacts@v4
+        id: artifacts
         with:
-          repo: metalk8s
-          input_path: metalk8s_sbom/repo/metalk8s
-          output_path: metalk8s_sbom/sbom
+          method: setup
+          url: https://artifacts.scality.net
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+
+      - name: Donwload artifacts
+        shell: bash
+        env:
+          ARTIFACTS_URL: ${{inputs.artifacts-url || steps.artifacts.outputs.link}}
+          ARTIFACTS_USER: ${{secrets.ARTIFACTS_USER}}
+          ARTIFACTS_PASSWORD: ${{secrets.ARTIFACTS_PASSWORD}}
```

Just for consistency

```suggestion
          ARTIFACTS_URL: ${{ inputs.artifacts-url || steps.artifacts.outputs.link }}
          ARTIFACTS_USER: ${{ secrets.ARTIFACTS_USER }}
          ARTIFACTS_PASSWORD: ${{ secrets.ARTIFACTS_PASSWORD }}
```